### PR TITLE
ENH: improve ARGUS rv generation in scipy.stats

### LINKED ISSUE #11699: Numerical issues in stats.argus for small shape parameters
Some methods of stats.argus do not work for very small values of the shape parameter chi.

#### Reproducing code example:

Example involving the cdf:

```
stats.argus.cdf(0.5, 1e-3) # 0.35048001182184085
stats.argus.cdf(0.5, 1e-4) # 0.3505843071786311
stats.argus.cdf(0.5, 1e-5) # 0.5
stats.argus.cdf(0.5, 1e-6) # nan
```

As chi goes to zero, the cdf converges to `1 - np.sqrt(1-x*x)**1.5`, so the limit is well-defined

```
cdf = lambda x: 1 - np.sqrt(1-x*x)**1.5
cdf(0.5) # 0.350480947161671
```

#### Error message:
The last line produces a warning:
```
RuntimeWarning: invalid value encountered in true_divide
  return _argus_phi(chi * np.sqrt(1 - x**2)) / _argus_phi(chi)
```

#### Scipy/Numpy/Python version information:
```
import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
1.3.0 1.15.4 sys.version_info(major=3, minor=7, micro=1, releaselevel='final', serial=0)
```

### PR DESCRIPTION
**Pull request by chrisb83**
#### Reference issue
Closes #11699

#### What does this implement/fix?
Follow-up PR to #11188 that improves performance of the ARGUS rvs method. Fixes numerical issues in `argus.rvs` if shape parameter chi is small

#### Additional information
The performance is improved for chi > 1.8 (direct transformation of Gamma rvs conditioned on [0, chi**2/2] replacing the RoU method). for chi < 0.5, a different rejection density is used to get stable results for very small values of chi (bug observed in referenced issue).

### CODE REVIEW

File: scipy/stats/tests/test_distributions.py
Comment (pvanmulbregt) [inline scipy/stats/tests/test_distributions.py:4341]: Perhaps candidates for consolidation under `pytest.mark.parametrize`  E.g. Something like
```
    @pytest.mark.parametrize('chi, random_state'), [
           [0.1, 325]   # chi < 1: rejection method
           [3.5, 135]   # for chi > 1.825, rvs transforms a conditional Gamma distribution
        ])
    def test_rvs(chi, random_state):
        x = stats.argus.rvs(chi, size=500, random_state= random_state)
         _, p = stats.kstest(x, "argus", (chi, ))
         assert_(p > 0.05)
```
.

Comment (chrisb83) [reply scipy/stats/tests/test_distributions.py:4677]: I changed it, thanks for the suggestion. I also rebased on master to update size and random_state treatment

**Comment by chrisb83** [comment]
I added functionality to generate rvs in Cython if `random_state` is a `numpy.random.Generator`. There are some problems:
- `numpy.random.Generator` cannot be used in Cython if the version of numpy is < 1.19 (https://github.com/numpy/numpy/pull/15463). After looking in more detail, it could be that 1.18 should be fine, I need to check [related question: SciPy-Dev Digest, Vol 195, Issue 13]
- I am not sure if 'numpy.random.RandomState' can be used efficiently in Cython [My related question is in SciPy-Dev Digest, Vol 195, Issue 25]. At the moment, I use pure Python code in that case which leads to redundant code.

File: scipy/stats/_continuous_distns.py
Comment (andyfaff) [inline scipy/stats/_continuous_distns.py:8504]: The default random_state is still a `RandomState`, not a `Generator`, and I'm guessing that won't change in the near future. Thus, the calls to this `_rvs` function *mostly* won't use this cython path. As such, is it worth adding the cython code?

File: scipy/stats/_stats.pyx
Comment (andyfaff) [inline scipy/stats/_stats.pyx:597]: Have you looked through the annotations of this cython code to see if there are Python dependent calls that can be removed?

**Review by andyfaff — COMMENTED** [review]
What are the relative timings of the old and new ways?

**Comment by mdhaber** [comment]
@andyfaff I just checkout it out:

%timeit argus.rvs(1, size=10000)
master 1.74 ms ± 113 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
PR 1.66 ms ± 42.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit argus.rvs(2.13, size=10000)
master 2.44 ms ± 574 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
PR 1.3 ms ± 28.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit argus.rvs(4.68, size=100000)
master 16.1 ms ± 209 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
PR 7.29 ms ± 63.7 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

%timeit argus.rvs(0.5, size=100000)
master 9.43 ms ± 707 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
PR 9.41 ms ± 189 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

**Comment by andyfaff** [comment]
My original review was just giving it a look over, rather that someone that knows the subject matter intimately. Don't wait for me if the PR looks suitable for a merge. Are the test fails fixed?

**Comment by chrisb83** [comment]
Hi Andy, Matt,

a few updates from my side on the performance improvement due to this PR:

- If Cython is not used, this PR should improve performance by relying on better rejection algorithms than before. I noted that the performance depends on the system. The thresholds to switch between the methods I proposed in the PR are not necessarily optimal and it seems advisable to focus on the methods that work best overall instead of trying to gain a little bit of performance on one system that does not materialize on another.
- if Cython is used, the speed-up compared to Cython is pretty substantial, at least for small values of chi but it depends again on the system, see an example below.

For 1mn rvs and chi=0.01, Cython vs Python: 
- 14ms vs 50ms on my relatively fast laptop with Linux
- 160ms vs 280ms on my slow machine with Win10

chi=2.0
- 41ms vs 64ms (Linux)
- 320 vs 416mn (Win10)

So I will try to simplify the code quite a bit, using just two methods each for Python and Cython. In my view, Cython then gives a decent improvement of the performance without adding too much redundant code + it serves as an example how to use Cython to generate rvs.

**Comment by chrisb83** [comment]
I ended up removing the Cython code. It creates too much overhead and speed is not that critical for such an "exotic" distribution. I hope the PR can be included in 1.8.0. One goal is to get rid of the dependency on `stats.rvs_ratio_uniforms` which is a candidate for deprecation.

**Comment by chrisb83** [comment]
To test the speed against the existing implementation, I just ran a few test cases on master. To generate 1mn rvs, the avg and std deviation of the generation times in ms over 20 trials are shown below:

chi=0.01: 60.1 +/- 12.4
chi=0.5: 63.3 +/- 10.3
chi=1: 63.7 +/- 9.4
chi=1.2: 64.5 +/- 9.0
chi=1.5: 64.9 +/- 8.6
chi=1.8: 65.8 +/- 8.5
chi=2: 66.9 +/- 8.7
chi=2.5: 70.0 +/- 13.5
chi=5: 74.8 +/- 19.0

Timings based on this PR to be added

**Comment by chrisb83** [comment]
I rebased the changes on master. From my point of view, this PR is ready.

**Comment by mdhaber** [comment]
I'll take a look.

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:24]: This and `from cpython.pycapsule cimport PyCapsule_IsValid, PyCapsule_GetPointer` below don't look related. Just some things you wanted to get done? If so, no problem. Just wanted to make sure they're intentional.

Comment (mdhaber) [reply scipy/stats/_continuous_distns.py:24]: Oops, this is related. How about they PyCapsule stuff, though?

File: scipy/stats/tests/test_distributions.py
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5784]:
```suggestion
            [0.1, 325],   # chi <= 1: rejection method case 1
```
right?

File: scipy/stats/tests/test_distributions.py
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5785]:
```suggestion
            [1.3, 155],   # 0.5 <= chi <= 1.8: rejection method case 2
```
right?

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:24]: The pycapsule import can be removed

File: scipy/stats/tests/test_distributions.py
Comment (chrisb83) [reply scipy/stats/tests/test_distributions.py:5785]: Correct

**Comment by chrisb83** [comment]
Thanks Matt, let me know if any other changes are required. To save CI resources, I will wait with the minor code updates

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:9138]: Where does this come from?

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:9149]: Is there a source, or did you figure this out? I'm not familiar with this sort of argument, so I don't know how to verify it.

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:9181]: The repetition here makes me wonder if there could be a handy private function to factor out for these two cases or maybe all three? If it's only used twice, copy-paste is OK, but let me know.

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:9134]: yes, correct. There is no reference, I will send you a note how to derive it.

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:9138]: image

I admit it is not obvious ;)

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:9134]: image

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:9149]: You can derive this as follows: if t is increasing, then P[t(X) <= u] = P[X <= t^{-1}(u)].
Now that derivatives and you get the relationship between the pdf of t(X) and X. I don't recommend checking the math :)

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:9181]: I considered that but given that you have different transformation got get from z to rvs, I felt it would not simplify it much and create some overhead, too

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:9131]: yes, f = self.pdf

you can find the book of Devroye here: http://www.eirene.de/Devroye.pdf

This is case 1:

image

**Comment by chrisb83** [comment]
Hi Matt,

thanks for looking into this and sorry for not providing more details about the math. I tried to keep it as short as possible, similar to the comment before, mainly to give the main idea.  I hope the test cases show that it works :) Do you think it is enough to document some more details in this PR? I can also share my notes on this with you.

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:9131]: and to answer your question:

your notation is f(x) <= M*g(x) where f and g
I am using a special case of that: f(x) = c g(x) h(x) with h in [0, 1], such that f(x) <= c g(x) => so indeed, c = M, it is well explained in Devroye

Comment (mdhaber) [reply scipy/stats/_continuous_distns.py:9131]: Where does this equation 9 come from? It's not in Devroye.

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [reply scipy/stats/_continuous_distns.py:9134]: So... there is a reference? Where does this equation 3 come from?

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [reply scipy/stats/_continuous_distns.py:9138]: Is this from the same source as the screenshots above or something else?

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [reply scipy/stats/_continuous_distns.py:9181]: > different transformation got get from z to rvs

You have written it with different transformations from z to RVS, but I think that just happens to be how the code is written. If you were to implement the formulae in the screenshots you included above, I think the function for performing this sort of rejection sampling would just need a formula for generating the sample `X(V)` and a rejection criterion (maybe some function of `U`, `X`, `V` is <= 0).

**Comment by mdhaber** [comment]
I made some histograms with this code:
```
import numpy as np
from scipy.stats import argus
from matplotlib import pyplot as plt

chi = 0.0001
dist = argus(chi)
N = 100000

x = np.linspace(0, 1, 300)
pdf = dist.pdf(x)
rvs = dist.rvs(N)

plt.plot(x, pdf)
plt.hist(rvs, density=True, bins=30)
plt.xlabel('x')
plt.ylabel('f(x)')
plt.legend(["PDF", "Histogram of RVS"])
plt.title(f"PDF vs RVS Histogram of ARGUS distribution w/ chi={chi}")
```
Here are some results:

<details>

image
image
image
image
image
image
image
image

</details>

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:9134]: It is a note that I have written, I can share it with you if you like

**Comment by chrisb83** [comment]
>  I don't see much performance difference otherwise, including chi=1.

Correct, for small chi, the new method might be a bit faster, but that is not a big difference. It is more accurate for very small chi.

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [reply scipy/stats/_continuous_distns.py:9134]: I think it would make it easier to verify the math, since I'm not familiar with it. Yes, please.

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:9138]:
```suggestion
        # g(x) = chi**2 * x * exp(-chi**2 * (1-x**2)/2) / (1 - exp(-chi**2 /2))
```

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:9129]:
```suggestion
        # write: PDF f = c * g(x) * h(x), where
```

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:9134]:
```suggestion
```
This note suggested to me that the sampling method relied on a small chi approximation. After careful review, I saw that this fact is not used in the method, so I think it can be removed from here.  If you want the fact to be emphasized, perhaps add a more direct unit test:

```
import numpy as np
from scipy.stats import argus

np.random.seed(0)
chi = 1e-9
dist = argus(chi)
rvs = dist.rvs(100)

np.random.seed(0)
np.random.uniform(size=100) # burn the samples for U
# alternatively, swap sampling of U and V in _rvs_scalar
V = np.random.uniform(size=100)
rvs2 = np.sqrt(1 - V**(2/3))

np.testing.assert_allclose(rvs, rvs2)
```

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:9143]:
```suggestion
        # In both cases, the inverse cdf of g can be written analytically, and
        # we can apply the rejection method:
        #
        # REPEAT
        #    Generate U uniformly distributed on [0, 1]
        #    Generate X with density g (e.g. via inverse transform sampling)
        # UNTIL X <= h(X)
        # RETURN X 
        #
        # We use case 1 for chi <= 0.5 due to its robustness to small chi
        # and case 2 for 0.5 < chi <= 1.8 due to its speed for moderate chi.
```
This little bit of extra context would have helped me understand the code.

File: scipy/stats/tests/test_distributions.py
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5794]: Consider a more direct test e.g. https://github.com/scipy/scipy/pull/11954/files#r754718823

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:9175]: Only g and h are written above. Even if the reader goes through the trouble of calculating G and its inverse, it is still a little extra work to verify that the code uses them because of the rearrangements done in the code. (For instance, in Case 1, we do inverse transform sampling using the SF instead of the PPF; in Case 2, the PPF is heavily refactored to avoid precision issues.) So I'd suggest adding more comments like this rather than removing them.

**Review by mdhaber — COMMENTED** [review]
In addition to the numerical checks [above](https://github.com/scipy/scipy/pull/11954#issuecomment-973352406), I've checked the math and code for Case 1 and Case 2 carefully.

For reference, here is my code for checking the math.
<details>
from IPython.display import display
from sympy import (symbols, pi, exp, sqrt, erf,
                   diff, Rational, limit, simplify,
                   integrate, solve, log)

x, chi, X, V = symbols('x, \chi, X, V', real=True, positive=True)

phi = 1/sqrt(2*pi)*exp(-chi**2/2)
Phi = (1+erf(chi/sqrt(2)))/2
Psi = Phi - chi*phi - Rational(1, 2)

f0 = chi**3/(sqrt(2*pi) * Psi)*x*sqrt(1-x**2)*exp(-chi**2*(1-x**2)/2)
c = chi**3/(3*sqrt(2*pi) * Psi)
g = 3*x*sqrt(1-x**2)
h = exp(-chi**2*(1-x**2)/2)
f1 = c * g * h

assert(f1.equals(f0))

G = 1 - (1-x**2)**(Rational(3, 2))
assert(G.subs(x, 0) == 0)
assert(G.subs(x, 1) == 1)
assert(simplify(diff(G, x) - g) == 0)

display(solve((1-V) - G, x)[0])

c = chi*(1-exp(-chi**2/2))/(sqrt(2*pi)*Psi)
g = chi**2 * x * exp(-chi**2*(1-x**2)/2)/(1-exp(-chi**2/2))
h = sqrt(1-x**2)
f2 = c * g * h
assert(f2.equals(f0))

G2 = (exp(chi**2*x**2/2) - 1)/(exp(chi**2/2)-1)
assert(G2.subs(x, 0) == 0)
assert(G2.subs(x, 1) == 1)
assert(simplify(diff(G2, x) - g) == 0)


G2inv0 = sqrt(1+2*log(exp(-chi**2/2)*(1-V)+V)/chi**2)
G2inv1 = solve(V - G2, x)[1]
G2inv2 = sqrt(2*log(V*(exp(chi**2/2)-1)+1)/chi**2)
assert(simplify(G2inv0 - G2inv1) == 0)
assert(simplify(G2inv0 - G2inv2) == 0)

display(G2inv0)
</details>

I checked numerical results by writing everything quite directly:

```python3
        def devroye_rejection(N, Ginv, h, random_state):
            x = np.zeros(N)
            simulated = 0
            while simulated < N:
                k = N - simulated
                U = random_state.uniform(size=k)
                V = random_state.uniform(size=k)
                X = Ginv(V)
                accept = (U <= h(X))
                num_accept = np.sum(accept)
                if num_accept > 0:
                    x[simulated:(simulated + num_accept)] = X[accept]
                    simulated += num_accept
            return x
```

The two cases become:
```python3
        if chi <= 0.5:
            def Ginv(V):
                # Technically this is the inverse of 1-G, the ISF, right?
                return np.sqrt(1 - V**(2/3))
            def h(X):
                return np.exp(-0.5* chi2 * (1 - X**2))
        else:
            def Ginv(V):
                # np.sqrt(2*np.log(V*(np.exp(chi2/2)-1)+1))/chi
                # refactored for precision with small chi
                return np.sqrt(1 + 2*np.log(np.exp(-chi2/2)*(1-V)+V)/chi2)
            def h(X):
                return np.sqrt(1 - X**2)
        x  = devroye_rejection(N, Ginv, h, random_state)
```
Presumably the way you have it is slightly faster? I'm OK keeping it as-is; just let me know why it should be kept that way (e.g computing `z = v**(2/3)`, computing a modified acceptance condition in terms of `z`, and then computing the actual random variates `rvs = np.sqrt(1 - z[accept]` instead of following the steps of the algorithm in order and using the natural form of expressions).

I still don't understand the Gamma stuff. You wrote a great note explaining the other cases; could you make the part about the relationship to the Gamma distribution were as complete and explicit so I can follow? Or are there any references that explain similar relationships that i could adapt to this case?

You [didn't recommend](https://github.com/scipy/scipy/pull/11954/files#r752567459) checking the math, but I hesitate to merge this part without a second opinion when there is nothing to check against but histograms and a KS test. I believe it's right; i just think that a second person should understand it.

File: scipy/stats/tests/test_distributions.py
Comment (chrisb83) [reply scipy/stats/tests/test_distributions.py:5794]: personally I find the existing test easier to understand: in your example, you correctly "burn" the first 100 samples when reusing the seed. you can do that since the rejection constant is almost equal to one, but that might not be obvious. if ok for you, I would keep the existing test

File: scipy/stats/tests/test_distributions.py
Comment (chrisb83) [reply scipy/stats/tests/test_distributions.py]: that is fixed

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:9131]: should be fixed after committing your suggestions:
- sampling method is better explained
- pdf=f is explained

**Comment by chrisb83** [comment]
Thanks a lot for the detailed review, Matt!

> Presumably the way you have it is slightly faster? I'm OK keeping it as-is; just let me know why it should be kept that way (e.g computing z = v**(2/3), computing a modified acceptance condition in terms of z, and then computing the actual random variates rvs = np.sqrt(1 - z[accept] instead of following the steps of the algorithm in order and using the natural form of expressions).

it is common the simplify the acceptance condition if possible to avoid some function evaluations, but let me double check if this makes sense here.

> I still don't understand the Gamma stuff. You wrote a great note explaining the other cases; could you make the part about the relationship to the Gamma distribution were as complete and explicit so I can follow? Or are there any references that explain similar relationships that i could adapt to this case?

was my handwritten note I sent you helpful? I can add that to the note with the PDF and attach it here. One can also check it with SymPy I guess?

File: scipy/stats/tests/test_distributions.py
Comment (mdhaber) [reply scipy/stats/tests/test_distributions.py:5794]: Sure, the existing test is a little easier to understand because you don't have to burn any samples. This test is just much stronger because it checks that the samples themselves are *correct* rather than that they "do not cause a null hypothesis to be rejected".

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [reply scipy/stats/_continuous_distns.py:9156]: This comment makes sense now that I have followed the math. Maybe add:
```suggestion
        # conditioned on [0, chi**2 / 2]). Therefore, to sample X from the
        # ARGUS distribution, we sample Y from the gamma distribution, keeping
        # only samples on [0, chi**2 / 2], and apply the inverse
        # transformation X = (1 - 2*Y/chi**2)**(1/2).
```

File: scipy/stats/_continuous_distns.py
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:9147]:
```suggestion
        # RETURN X
```
To fix lint error: `scipy/stats/_continuous_distns.py:9147:19: W291 trailing whitespace`

**Review by mdhaber — APPROVED** [review]
Thanks for chatting today. Now that I've run through the relationship between ARGUS and the Gamma distribution, I think everything looks good. There are a few places I've commented before that were still under discussion; just let me know your final thoughts, make any last adjustments, and we can merge.

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py]: done

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:9175]: done

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py]: done

**Comment by chrisb83** [comment]
> Presumably the way you have it is slightly faster? I'm OK keeping it as-is; just let me know why it should be kept that way (e.g computing z = v**(2/3), computing a modified acceptance condition in terms of z, and then computing the actual random variates rvs = np.sqrt(1 - z[accept] instead of following the steps of the algorithm in order and using the natural form of expressions).

I checked this again and it makes sense to keep it. The main benefit is performance:
- you first check the simplified acceptance condition U <= h(G_inv(V)) => often this can be done w/o evaluating all expressions in G_inv and h by rewriting the condition
- only if the acceptance condition is fulfilled, you apply the final transformation. in the cited text, this saves the evaluation of sqrt on the samples that are rejected

I added more comments inline now.

File: scipy/stats/_continuous_distns.py
Comment (chrisb83) [reply scipy/stats/_continuous_distns.py:9203]: see explanation in https://github.com/scipy/scipy/pull/11954#issuecomment-981938909

**Comment by mdhaber** [comment]
Updates look fine. Let's just confirm that the doc build failures are not from this PR? They don't look related, but I'm not seeing the issue in other PRs. I'm going to try merging master.

**Comment by chrisb83** [comment]
> Update #2: I guess it's master. See gh-15119.

Yes

**Comment by mdhaber** [comment]
OK, ready to merge, then?

**Comment by chrisb83** [comment]
> OK, ready to merge, then?

Yes, I don't plan to change anything else. Thanks

**Comment by mdhaber** [comment]
Thanks @chrisb83!

**Pull request merged**
